Sources/ATAChat/Managers/ChannelController.swift

```diff
@@ -35,60 +35,41 @@ public class ChannelController {
             return
         }
-        guard let passenger = ride.passenger else {
-            return
-        }
-//        userReference.whereField("id", isEqualTo: passenger.id).addSnapshotListener { (querySnapshot, error) in
-//                    guard let documents = querySnapshot?.documents else {
-//                        print("No documents")
-//                        return
-//                    }
-//
-//                    let users = documents.map { (queryDocumentSnapshot) -> ChatUserT in
-//                        let docId = queryDocumentSnapshot.reference.documentID
-//                        let data = queryDocumentSnapshot.data()
-//                        let name = data["name"] as? String ?? ""
-//                        let id = data["id"] as? Int ?? 0
-//                        let chatUser: ChatUserT = ChatUserT(id: id, name: name)
-//                        return chatUser
-//                    }
-//                    print(users)
-//                }
         
-        getChatIds(for: [driver.id, passenger.id]) { ids in
-            guard let ids = ids else {
-                return
-            }
-            guard let driverChatId = ids[driver.id], let passengerChatId = ids[passenger.id] else {
-                return
+        guard driver.chatId != "", passenger.chatId != "" else {
+            getChatIds(for: [driver.id, passenger.id]) { [weak self] ids in
+                guard let ids = ids else {
+                    return
+                }
+                guard let driverChatId = ids[driver.id], let passengerChatId = ids[passenger.id] else {
+                    return
+                }
+                self?.createRideChannel(for: ride, with: driverChatId, and: passengerChatId)
             }
-            let channelName = "Course du \(ChannelController.chanelNameDateFormatter.string(from: ride.ride.startDate.value))"
-            let createdAt = ChannelController.createdDateFormatter.string(from: Date())
-            let channelId = [driverChatId, "#", passengerChatId].joined()
-            let data: [String:Any] = ["name": channelName, "user": [driverChatId, passengerChatId], "createdAt":createdAt]
-            self.channelReference.document(channelId).setData(data)
+            return
         }
-        
-        
-        
-        //let data: [String:Any] = ["name": channelName, "user": [driver.chatId, passenger.chatId]]
-        //channelReference.document([driver.chatId, "#", passenger.chatId].joined()).setData(data)
-        
-//        let tmp = userReference.whereField("id", arrayContains: "\(passenger.id)").getDocuments{a, b in
-//            print(a)
-//            print(b)
-//        }
-//        print(tmp)
+        createRideChannel(for: ride, with: driver.chatId, and: passenger.chatId)
+    }
+    
+    private func createRideChannel(for ride: OngoingRide, with driverChatId: String, and passengerChatId: String){
```

Comment:
with et and sont assez obscures en lisant, utilise plutôt driverChatId et passengerChatId peut être ?